crates/catalog/sql/src/catalog.rs

```diff
@@ -0,0 +1,836 @@
+                })
+            })
+            .collect::<std::result::Result<_, sqlx::Error>>()
+            .map_err(from_sqlx_error)?)
+    }
+
+    async fn table_exists(&self, identifier: &TableIdent) -> Result<bool> {
+        let catalog_name = self.name.clone();
+        let namespace = identifier.namespace().join(".");
+        let name = identifier.name().to_string();
+        let rows = self
+            .execute_statement(
+                &format!("select {}, {}, {}, {} from {} where {} = ? and {} = ? and {} = ? and ({} = '{}' or {} is null);", 
+                    TABLE_NAMESPACE,
+                    TABLE_NAME,
+                    METADATA_LOCATION_PROP,
+                    PREVIOUS_METADATA_LOCATION_PROP,
+                    CATALOG_TABLE_VIEW_NAME,
+                    CATALOG_NAME,
+                    TABLE_NAMESPACE,
+                    TABLE_NAME,
+                    RECORD_TYPE,
+                    TABLE_RECORD_TYPE,
+                    RECORD_TYPE),
+                vec![Some(&catalog_name), Some(&namespace), Some(&name)],
+            )
+            .await?;
+        let mut iter = rows.iter().map(query_map);
+
+        Ok(iter.next().is_some())
+    }
+
+    async fn drop_table(&self, identifier: &TableIdent) -> Result<()> {
+        let catalog_name = self.name.clone();
+        let namespace = identifier.namespace().to_url_string();
+        let name = identifier.name.to_string();
+
+        self.execute_statement(
+            &format!(
+                "delete from {} where {} = ? and {} = ? and {} = ?",
+                CATALOG_TABLE_VIEW_NAME, CATALOG_NAME, TABLE_NAMESPACE, TABLE_NAME
+            ),
+            vec![Some(&catalog_name), Some(&namespace), Some(&name)],
+        )
+        .await?;
+
+        Ok(())
+    }
+
+    async fn load_table(&self, identifier: &TableIdent) -> Result<Table> {
+        let metadata_location = {
+            let catalog_name = self.name.clone();
+            let namespace = identifier.namespace().to_url_string();
+            let name = identifier.name().to_string();
+            let row = self
+                .execute_statement(
+                    &format!("select {}, {}, {}, {} FROM {} where {} = ? and {} = ? and {} = ? and ({} = '{}' or {} is null)", 
+                        TABLE_NAMESPACE,
+                        TABLE_NAME,
+                        METADATA_LOCATION_PROP,
+                        PREVIOUS_METADATA_LOCATION_PROP,
+                        CATALOG_TABLE_VIEW_NAME,
+                        CATALOG_NAME,
+                        TABLE_NAMESPACE,
+                        TABLE_NAME,
+                        RECORD_TYPE,
+                        TABLE_RECORD_TYPE,
+                        RECORD_TYPE
+                    ),
+                    vec![Some(&catalog_name), Some(&namespace), Some(&name)],
+                )
+                .await?;
+            assert_eq!(row.len(), 1, "expected only one row from load_table query");
```

Comment:
If this assertion fails, the service panics and exits. A more graceful failure would be desirable.

Reply:
this is a good point, in hindsight the table's primary key should stop this even happening so I have removed.